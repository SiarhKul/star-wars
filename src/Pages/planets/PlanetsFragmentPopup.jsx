import React from "react";
import { useSelector } from "react-redux";
import { PopupListItem } from "../../components/popup/PopupListItem";

export const PlanetsFragmentPopup = () => {
	const {
		name,
		rotation_period,
		diameter,
		climate,
		gravity,
		terrain,
		surface_water,
		population,
		residents,
		films,
	} = useSelector(state => state.app.clickedCard);

	return (
		<>
			<div className="name-wrapper">
				<span className="popup__name">{name}</span>
			</div>
			<div className="description-wrapper">
				<div className="description-person">
					<div className="description-appearance">
						<p className="description__heading">Appearance</p>
						<ul className="description-appearance__item">
							<li>Rotation period:{rotation_period}</li>
							<li>Period period:{rotation_period}</li>
							<li>Diameter:{diameter}</li>
							<li>Climate:{climate}</li>
							<li>Gravity:{gravity}</li>
							<li>Terrain:{terrain}</li>
							<li>Surface water:{surface_water}</li>
							<li>Population:{population}</li>
						</ul>
					</div>
				</div>

				<div className="description-resources">
					<div className="resources">
						<div className="resources-heading">Residents:</div>
						<ul className="resources-list">
							<PopupListItem urls={residents} name="name" path="" />
						</ul>
					</div>

					<div className="resources">
						<div className="resources-heading">Films:</div>
						<ul className="resources-list">
							<PopupListItem urls={films} name="title" path="films" />
						</ul>
					</div>
				</div>
			</div>
		</>
	);
};
