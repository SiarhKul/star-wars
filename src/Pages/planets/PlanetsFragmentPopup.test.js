import { shallow } from 'enzyme';
import React from 'react';
import { PlanetsFragmentPopup } from './PlanetsFragmentPopup';

jest.mock('history', () => ({
  createBrowserHistory: jest.fn(() => ({
    location: {
      state: {
        name: 'Tatooine',
        rotation_period: '23',
        diameter: '10465',
        climate: 'arid',
        gravity: '1 standard',
        terrain: 'desert',
        surface_water: '1',
        population: '200000',
        residents: ['https://swapi.dev/api/people/1/'],
        films: ['https://swapi.dev/api/films/1/'],
      },
    },
  })),
}));

describe('Test <PlanetsFragmentPopup/>', () => {
  const planetsFragmentPopup = () => shallow(<PlanetsFragmentPopup />);
  const component = planetsFragmentPopup();

  it('render snapshot <PlanetsFragmentPopup/>', () => {
    expect(component).toMatchSnapshot();
  });

  it('should  <PlanetsFragmentPopup/> has required fields', () => {
    const requiredFields = [
      'Appearance',
      'Rotation period:',
      'Period period:',
      'Diameter:',
      'Climate:',
      'Gravity:',
      'Terrain:',
      'Surface water:',
      'Population:',
      'Residents:',
      'Films:',
    ];
    expect(component.containsAllMatchingElements(requiredFields)).toBeTruthy();
  });
});
