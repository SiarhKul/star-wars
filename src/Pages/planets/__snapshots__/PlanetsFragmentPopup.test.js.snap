// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PlanetsFragmentPopup/> render snapshot <PlanetsFragmentPopup/> 1`] = `
<Fragment>
  <div
    className="name-wrapper"
  >
    <span
      className="popup__name"
    >
      Tatooine
    </span>
  </div>
  <div
    className="description-wrapper"
  >
    <div
      className="description-person"
    >
      <div
        className="description-appearance"
      >
        <p
          className="description__heading"
        >
          Appearance
        </p>
        <ul
          className="description-appearance__item"
        >
          <li>
            Rotation period:
            23
          </li>
          <li>
            Period period:
            23
          </li>
          <li>
            Diameter:
            10465
          </li>
          <li>
            Climate:
            arid
          </li>
          <li>
            Gravity:
            1 standard
          </li>
          <li>
            Terrain:
            desert
          </li>
          <li>
            Surface water:
            1
          </li>
          <li>
            Population:
            200000
          </li>
        </ul>
      </div>
    </div>
    <div
      className="description-resources"
    >
      <div
        className="resources"
      >
        <div
          className="resources-heading"
        >
          Residents:
        </div>
        <ul
          className="resources-list"
        >
          <PopupListItem
            name="name"
            path="people"
            urls={
              Array [
                "https://swapi.dev/api/people/1/",
              ]
            }
          />
        </ul>
      </div>
      <div
        className="resources"
      >
        <div
          className="resources-heading"
        >
          Films:
        </div>
        <ul
          className="resources-list"
        >
          <PopupListItem
            name="title"
            path="films"
            urls={
              Array [
                "https://swapi.dev/api/films/1/",
              ]
            }
          />
        </ul>
      </div>
    </div>
  </div>
</Fragment>
`;
